From: Matthew Wilcox <matthew.r.wilcox@intel.com>
Date: Mon, 19 Sep 2011 17:14:53 -0400
Subject: NVMe: Fix memory leak in nvme_dev_add()
Git-commit: 684f5c2025b067a23722e620d0b3b858d8dc5d01
References: FATE#313627
Patch-Mainline: v3.7-rc1

The driver was allocating 8k of memory, then freeing 4k of it.

Signed-off-by: Matthew Wilcox <matthew.r.wilcox@intel.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/block/nvme.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/block/nvme.c b/drivers/block/nvme.c
index a7f82fb..705f66e 100644
--- a/drivers/block/nvme.c
+++ b/drivers/block/nvme.c
@@ -1457,7 +1457,7 @@ static int __devinit nvme_dev_add(struct nvme_dev *dev)
 	list_for_each_entry(ns, &dev->namespaces, list)
 		add_disk(ns->disk);
 
-	dma_free_coherent(&dev->pci_dev->dev, 4096, id, dma_addr);
+	dma_free_coherent(&dev->pci_dev->dev, 8192, id, dma_addr);
 	return 0;
 
  out_free:
@@ -1466,7 +1466,7 @@ static int __devinit nvme_dev_add(struct nvme_dev *dev)
 		nvme_ns_free(ns);
 	}
 
-	dma_free_coherent(&dev->pci_dev->dev, 4096, id, dma_addr);
+	dma_free_coherent(&dev->pci_dev->dev, 8192, mem, dma_addr);
 	return res;
 }
 
-- 
1.7.4.2

